packages/big-design/src/components/PillTabs/PillTabs.tsx

```diff
@@ -1,97 +1,93 @@
 
 import { StyledFlexItem, StyledPillTab } from './styled';
+import { useAvailableWidth } from './useAvailableWidth';
 
 export interface PillTabItem {
   id: string;
   title: string;
 }
 
+interface Pill extends PillTabItem {
+  isVisible: boolean;
+  ref: React.RefObject<HTMLDivElement>;
+}
+
 export interface PillTabsProps {
   items: PillTabItem[];
   activePills: string[];
   onPillClick: (itemId: string) => void;
 }
 
 export const PillTabs: React.FC<PillTabsProps> = ({ activePills, items, onPillClick }) => {
-  const parentRef = createRef<HTMLDivElement>();
-  const dropdownRef = createRef<HTMLDivElement>();
-  const [isMenuVisible, setIsMenuVisible] = useState(false);
-  const [pillsState, setPillsState] = useState(
-    items.map((item) => ({
-      isVisible: true,
-      item,
-      ref: createRef<HTMLDivElement>(),
-    })),
-  );
-
-  const hideOverflowedPills = useCallback(() => {
-    const parentWidth = parentRef.current?.offsetWidth;
-    const dropdownWidth = dropdownRef.current?.offsetWidth;
-
-    if (!parentWidth || !dropdownWidth) {
-      return;
-    }
-
-    let remainingWidth = parentWidth;
-
-    const newState = pillsState.map((stateObj) => {
-      const pillWidth = stateObj.ref.current?.offsetWidth;
-
-      if (!pillWidth) {
-        return stateObj;
-      }
-
-      if (remainingWidth - pillWidth > dropdownWidth) {
-        remainingWidth -= pillWidth;
-
-        return {
-          ...stateObj,
-          isVisible: true,
-        };
-      }
+  const availableWidth = useAvailableWidth();
 
-      return {
-        ...stateObj,
-        isVisible: false,
-      };
-    });
+  const itemsWithRefs = useMemo(
+    () => items.map((item) => ({ ...item, isVisible: true, ref: createRef<HTMLDivElement>() })),
+    [items],
+  );
 
-    const visiblePills = pillsState.filter((stateObj) => stateObj.isVisible);
-    const newVisiblePills = newState.filter((stateObj) => stateObj.isVisible);
+  const { pills } = itemsWithRefs.reduce<{ pills: Pill[]; widthBudget: number }>(
+    (acc, pill) => {
+      const pillWidth = pill.ref.current?.offsetWidth || 0;
+      const widthBudget = acc.widthBudget - pillWidth;
+      const updatedPill = { ...pill, isVisible: widthBudget >= 0 };
 
-    if (visiblePills.length !== newVisiblePills.length) {
-      setIsMenuVisible(newVisiblePills.length !== items.length);
-      setPillsState(newState);
-    }
-  }, [items, parentRef, dropdownRef, pillsState]);
+      return { pills: [...acc.pills, updatedPill], widthBudget };
+    },
+    { pills: [], widthBudget: availableWidth.value },
+  );
 
-  const renderedDropdown = useMemo(() => {
-    const dropdownItems = pillsState
-      .filter((stateObj) => !stateObj.isVisible)
-      .map((stateObj) => {
-        const item = items.find(({ title }) => title === stateObj.item.title);
-        const isActive = item ? activePills.includes(item.id) : false;
+  const dropdownItems = pills
+    .filter(({ isVisible }) => !isVisible)
```

Comment:
🤔 why this `isVisible` indirection? I guess I was expecting two lists to be produced.

```tsx
const { pills, dropdownItems } = itemsWithREfs.reduce();
```

Reply:
I _think_ this was originally there for consistency with accessibility, while things visually move about. It may also be needed for the mechanism of working out sizes. 

Each `PillTab` remains in the same horizontal parent, regardless of it also being in the dropdown. They are visually hidden after being copied over.

I didn't want to change the underlying HTML or styling as part of this refactoring, just incase that became more of a breaking change. E.g. people relying on the current accessibility setup, or people's tests expecting all buttons to be available without having to check if they're in the dropdown